conf/openshift/openshift.json

```diff
@@ -222,15 +227,17 @@
       }
     },
     {
-      "kind": "DeploymentConfig",
-      "apiVersion": "v1",
+      "kind": "StatefulSet",
+      "apiVersion": "apps/v1beta1",
       "metadata": {
         "name": "dataverse-postgresql",
         "annotations": {
           "template.alpha.openshift.io/wait-for-ready": "true"
         }
       },
       "spec": {
+        "serviceName" : "dataverse-postgresql-service",
+        "replicas" : 1,
```

Comment:
@danmcp raised the question of whether running a setup with only a master and no slaves would have any side effects. I think there would be no side effects and it would be similar to just running a single instance Postgres setup.

Our code is based off [this example](https://github.com/sclorg/postgresql-container/tree/master/examples/replica#deploymentconfig-postgresql-master), which states: 

> Once the master is started, it works as a standalone database server, fully independent of the slaves.

I dug in a little more to the image to see what is happening:
For the postgres image, the only side effect for running as master is that these configuration settings are copied into the main postgres config file: https://github.com/sclorg/postgresql-container/blob/master/src/root/usr/share/container-scripts/postgresql/openshift-custom-postgresql-replication.conf.template

To me, the above settings look innocuous. 

For further context, here is the master-binary: https://github.com/sclorg/postgresql-container/blob/master/src/root/usr/bin/run-postgresql-master

And that environment variable is called in this common script, which just copies the first file above into the main config settings: https://github.com/sclorg/postgresql-container/blob/b737ad26db4f8769e775ca165284f62c7f2a66db/src/root/usr/share/container-scripts/postgresql/common.sh

Reply:
@patrickdillon thanks, I just moved this to QA.